src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt

```diff
@@ -0,0 +1,33 @@
+package com.openlattice.mechanic.upgrades
+
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.organizations.processors.OrganizationEntryProcessor
+import org.slf4j.LoggerFactory
+
+class ClearJSONOrganizationRoles(private val toolbox: Toolbox) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(ClearJSONOrganizationRoles::class.java)
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2020_01_29.value
+    }
+
+    override fun upgrade(): Boolean {
+
+        val organizations = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+
+        logger.info("About to clear roles for organizations")
+
+        organizations.executeOnEntries(OrganizationEntryProcessor {
+            it.roles.clear()
```

Comment:
Is the plan to also delete roles field from Organization class?

Reply:
Not right now -- but when reading out an organization it's always populated by loading the roles through SecurePrincipalsManager and then calling `organization.roles.addAll( ... )` which leads to problems when that original set of roles is never updated or maintained (this is what I was referencing yesterday in hardcore btw)

Reply:
Thanks for clarifying that!

Reply:
np!